scylla-server/src/proxy/proxy-client.ts

```diff
@@ -1,21 +1,48 @@
 // Ignoring this because it wont build on github for some reason
 // @ts-ignore
-import { ErrorWithReasonCode, IConnackPacket, MqttClient } from 'mqtt/*';
-import { ServerMessage } from '../odyssey-base/src/types/message.types';
+import { ErrorWithReasonCode, IConnackPacket, IPublishPacket, MqttClient } from 'mqtt/*';
 import { Topic } from '../odyssey-base/src/types/topic';
-
+import { Run } from '@prisma/client';
+import NodeService from '../odyssey-base/src/services/nodes.services';
+import RunService from '../odyssey-base/src/services/runs.services';
+import DataService from '../odyssey-base/src/services/data.services';
+import DataTypeService from '../odyssey-base/src/services/dataTypes.services';
+import LocationService from '../odyssey-base/src/services/locations.services';
+import DriverService from '../odyssey-base/src/services/driver.services';
+import SystemService from '../odyssey-base/src/services/systems.services';
+import ProxyServer from './proxy-server';
+import { ClientData } from '../utils/message.utils';
+import { ServerData, ServerMessage } from '../odyssey-base/src/types/message.types';
 /**
  * Handler for receiving messages from Siren
  */
 export default class ProxyClient {
   connection: MqttClient;
+  // should a new run be created, based off if new connection & data received
+  createNewRun: boolean;
+  // storing run for the current connection, at start is undefined
+  currentRun: Run | undefined;
+
+  proxyServers: ProxyServer[];
+
+  recentLatitude: number | undefined;
+  recentLongitude: number | undefined;
+  recentRadius: number | undefined;
+  recentLocationName: string | undefined;
+  newLocation: boolean = true;
 
   /**
    * Constructor
    * @param socket The socket to send and receive messages from
    */
-  constructor(mqttClient: MqttClient) {
+  constructor(mqttClient: MqttClient, proxyServers: ProxyServer[]) {
     this.connection = mqttClient;
+    // only true first time after connected and data received
```

Comment:
these comments aren't great. They should be clearer/ be full thoughts if they're gonna be here